sycl/test/atomic_ref/add.cpp

```diff
@@ -181,12 +195,31 @@ int main() {
   }
 
   constexpr int N = 32;
+  // CHECK-LLVM: declare dso_local spir_func i32
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicIAdd
+  // CHECK-LLVM-SAME: (i32 addrspace(1)*, i32, i32, i32)
   add_test<int>(q, N);
```

Comment:
I'm curious what if the function isn't inlined? Is its inlining forced somehow?

Reply:
I believe that inlining can't take place as such. These `__spirv_Atomic*` functions are generated due to being included from [here](https://github.com/intel/llvm/blob/7a8e02a1e26ee6160c112581d8851272e35e4dc9/sycl/include/CL/__spirv/spirv_ops.hpp#L57), however no definition is provided. These external calls get lowered at later stages (e.g. mapped onto corresponding SPIR-V operands during translation from LLVM IR - see https://github.com/intel/llvm/pull/2765 as a fresh example).